base/Enums.jl

```diff
@@ -7,6 +7,11 @@ export Enum, @enum
 
 function basetype end
 
+"""
+    Enum
```

Comment:
Maybe useful to show that it is parametric, e.g.
```suggestion
    Enum{T<:Integer}
```